vllm/model_executor/models/llama.py

```diff
@@ -629,6 +644,14 @@ def permute(w: torch.Tensor, n_heads: int):
 
         return name, loaded_weight
 
+    def post_init_bnb_attrs(self):
+        self.fused_weights_modules = {
+            k: v
+            for submodule in self.modules()
+            if hasattr(submodule, "fused_weights_modules")
+            for k, v in submodule.fused_weights_modules.items()
+        }
```

Comment:
Perhaps we can use a similar logic for adding BNB support for VLMs that use arbitrary vision/language backbone in future PR?

For example, get BNB attrs from ViT and LLM backbone respectively, and combine them then.

Reply:
Sure. The implementation is clean and indeed significantly decrease static variable! With this implementation, we can remove `fused_weights_modules` variable added in this PR. 

(TBH, I don't really like the `fused_weights_modules` I added in this PR XD)

Reply:
This is a nice improvement, yeah sorry for the silence on this PR. I kept looking at it and was displeased with the model changes, so anything we can do to avoid additions to the model would be preferred